homeassistant/components/api.py

```diff
@@ -173,7 +174,17 @@ def forward_events(event):
 
 def _handle_get_api_config(handler, path_match, data):
     """Return the Home Assistant configuration."""
-    handler.write_json(handler.server.hass.config.as_dict())
+    if handler.authenticated:
+        handler.write_json(handler.server.hass.config.as_dict())
+    else:
+        needs_auth = (handler.server.hass.config.api.api_password is not None)
+        params = {
```

Comment:
I'm going to do this once this PR gets merged because UPnP has already been merged in.